### ISSUE TITLE
An unexpected false positive in MirChecker

### ISSUE THREAD

**Issue by soyo114** [lizhuohua/rust-mir-checker#21]
The overflow detection module in MirChecker may produce incorrect results when analyzing `if-else` code blocks, as it reports an obvious false positive. Below is a minimized code example demonstrating the issue.

## Code Example
```rust
fn fib(n: isize) -> isize {
    if n <= 1 {
        1
    } else {
        (n-1)
    }
}
fn main() {
    fib(isize::MAX);
}
```

## Analysis Results
MirChecker reports the following false positive:
```bash
warning: [MirChecker] Possible error: "attempt to compute `\"move _4\" - \"const 1_isize\"`, which would overflow"
 --> src/main.rs:5:9
  |
5 |         (n-1)
  |         ^^^^^
```
